flang/lib/Optimizer/Builder/Runtime/Stop.cpp

```diff
@@ -20,3 +21,18 @@ void fir::runtime::genExit(fir::FirOpBuilder &builder, mlir::Location loc,
       fir::runtime::createArguments(builder, loc, exitFunc.getType(), status);
   builder.create<fir::CallOp>(loc, exitFunc, args);
 }
+
+void fir::runtime::genCrash(fir::FirOpBuilder &builder, mlir::Location loc,
```

Comment:
Would it help to have a name that includes "message", like `genCrashMessage`?

Reply:
It's not really a crash.  It's a user error.

genReportUserError() ?

@psteinfeld The name in the runtime -- Crash() -- will show up in the user's traceback.  Maybe we could change it too?

Reply:
Thanks @sscalpone, your point makes sense to me. I would just keep a `Fatal` somewhere to make it clear the program will not continue. Maybe genReportFatalUserError  (I have updated this PR to that, but not the runtime) ?

Reply:
See https://reviews.llvm.org/D118703.